services/QuillLMS/app/services/demo/report_demo_creator.rb

```diff
@@ -292,19 +290,24 @@ def self.non_demo_classrooms(teacher)
     teacher.unscoped_classrooms_i_teach.reject {|c| c.code == classcode(teacher.id) }
   end
 
-  def self.create_units(teacher)
-    ACTIVITY_PACKS_TEMPLATES.map do |ap|
+  def self.create_units(teacher, is_teacher_demo)
+    ACTIVITY_PACKS_TEMPLATES.map do |activity_pack_template|
       # the following line sets the unit template id to nil for the quill_staff_demo account by request of the partnerships team, because they want to be able to assign the starter baseline recommendations
       # and it ensures the unit template actually exists in our database
-      unit_template_id = teacher.email == STAFF_DEMO_EMAIL ? nil : UnitTemplate.find_by_id(ap[:unit_template_id])&.id
-      unit = Unit.find_or_create_by(name: ap[:name], user: teacher, unit_template_id: unit_template_id)
-      activity_ids = activity_ids_for_config(ap)
-      activity_ids.each { |act_id| UnitActivity.find_or_create_by(activity_id: act_id, unit: unit) }
+      unit_template_id = is_teacher_demo ? UnitTemplate.find_by_id(activity_pack_template[:unit_template_id])&.id : nil
+      name = unit_name(activity_pack_template[:name], is_teacher_demo)
+      unit = Unit.find_or_create_by(name:, user: teacher, unit_template_id:)
+      activity_ids = activity_ids_for_config(activity_pack_template)
+      activity_ids.each { |activity_id| UnitActivity.find_or_create_by(activity_id:, unit:) }
 
       unit
     end
   end
 
+  def self.unit_name(name, is_teacher_demo)
+    is_teacher_demo ? name : "#{name} (Staff Demo)"
```

Comment:
My only concern here is that the staff demo isn't the only time we pass `is_teacher_demo: false` to this file. Right now, we do it for the admin demo reports too. I might actually be changing this in an upcoming PR, but I think it might be better to just do `(Demo)` here to avoid confusion in the future?